src/library/autodj/autodjprocessor.cpp

```diff
@@ -586,6 +587,20 @@ bool AutoDJProcessor::removeTrackFromTopOfQueue(TrackPointer pTrack) {
     if (m_pConfig->getValueString(ConfigKey(kConfigKey, "Requeue")).toInt()) {
         m_pAutoDJTableModel->appendTrack(nextId);
     }
+
+    // Fill random tracks if configured
+    int minAutoDJCrateTracks = m_pConfig->getValueString(
+            ConfigKey(kConfigKey, "RandomQueueMinimumAllowed")).toInt();
+    bool randomQueueEnabled = (((m_pConfig->getValueString(
+            ConfigKey("[Auto DJ]", "EnableRandomQueue")).toInt())) == 1);
+
+    bool minTracksLeft = (m_pAutoDJTableModel->rowCount() <= minAutoDJCrateTracks);
+
+    if (randomQueueEnabled && minTracksLeft) {
+        qDebug() << "Randomly adding tracks";
+        emit(randomTrackRequested(minTracksLeft));
```

Comment:
What about the cases where the user sets the minimum tracks as say 20 and starts off with 30 tracks in the autoDJ playlist. This condition will hold the random track addition till only 20 tracks are left .At that point only , will tracks be added randomly